### PR TITLE
copt deduplicate: render_dpi, margins, rotation_mode

### PR DESCRIPTION
**Pull request by hius07**
Main discussion in https://github.com/koreader/koreader/pull/10763.

The last round for ReaderTypeset.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10880)

### CODE REVIEW

**Comment by hius07** [comment]
Done with ReaderView.

File: frontend/ui/data/creoptions.lua [hunk 1]
```diff
@@ -51,7 +51,7 @@ local CreOptions = {
     {
         icon = "appbar.rotation",
         options = {
-            {
+            {   -- ReaderView
                 name = "rotation_mode",
                 name_text = _("Rotation"),
```
Comment (poire-z) [inline frontend/ui/data/creoptions.lua:56]: Good idea these comments :)

**Review by poire-z — APPROVED** [review]

File: frontend/apps/reader/modules/readerview.lua [hunk 2]
```diff
@@ -867,20 +867,12 @@ function ReaderView:onReadSettings(config)
...
-    end
-    if rotation_mode then
+    if G_reader_settings:nilOrFalse("lock_rotation") then
+        local setting_name = self.ui.paging and "kopt_rotation_mode" or "copt_rotation_mode"
+        -- document.configurable.rotation_mode is not ready yet
+        local rotation_mode = config:readSetting(setting_name)
+                           or G_reader_settings:readSetting(setting_name)
+                           or Screen.DEVICE_ROTATED_UPRIGHT
         self:onSetRotationMode(rotation_mode)
     end
```
[the remark this answers is not in the document]
Comment (NiLuJe) [reply frontend/apps/reader/modules/readerview.lua:877]: I lost track of things, but if the DocSetting variant is now actually prefixed like the global one, that looks good ;).

[the remark this answers is not in the document]
Comment (NiLuJe) [reply frontend/apps/reader/modules/readerview.lua:877]: > I lost track of things

The thing being we were actually saving both a prefixed and unprefixed variant in the SDR, and now we only keep the prefixed one, right?

[the remark this answers is not in the document]
Comment (hius07) [reply frontend/apps/reader/modules/readerview.lua:877]: Yes, only the prefixed one, that is saved by
https://github.com/koreader/koreader/blob/5e74f29fba143e7eb76eacecea8a55f13ef3cbbc/frontend/configurable.lua#L61-L69

File: frontend/apps/reader/modules/readertypeset.lua [hunk 3]
```diff
@@ -52,41 +52,27 @@ function ReaderTypeset:onReadSettings(config)
...
-                  or G_defaults:readSetting("DCREREADER_CONFIG_B_MARGIN_SIZES_LARGE")
-    self.unscaled_margins = { h_margins[1], t_margin, h_margins[2], b_margin }
+    self.unscaled_margins = { self.configurable.h_page_margins[1], self.configurable.t_page_margin,
+                              self.configurable.h_page_margins[2], self.configurable.b_page_margin }
     self:onSetPageMargins(self.unscaled_margins)
-    self.sync_t_b_page_margins = config:readSetting("copt_sync_t_b_page_margins")
-                              or G_reader_settings:readSetting("copt_sync_t_b_page_margins")
-                              or 0
-    self.sync_t_b_page_margins = self.sync_t_b_page_margins == 1 and true or false
+    self.sync_t_b_page_margins = self.configurable.sync_t_b_page_margins == 1 and true or false
```
[the remark this answers is not in the document]
Comment (hius07) [reply frontend/apps/reader/modules/readertypeset.lua:70]: https://github.com/koreader/koreader/blob/5e74f29fba143e7eb76eacecea8a55f13ef3cbbc/frontend/ui/data/creoptions.lua#L266
https://github.com/koreader/koreader/blob/5e74f29fba143e7eb76eacecea8a55f13ef3cbbc/frontend/configurable.lua#L40

[the remark this answers is not in the document]
Comment (hius07) [reply frontend/apps/reader/modules/readertypeset.lua:70]: And `DCREREADER_CONFIG_T_MARGIN_SIZES_LARGE` is equal to `DCREREADER_CONFIG_B_MARGIN_SIZES_LARGE`.

**Pull request merged**